curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641da615af82bf454215a992.md

```diff
@@ -0,0 +1,164 @@
+---
+id: 641da615af82bf454215a992
+title: Step 10
+challengeType: 0
+dashedName: step-10
+---
+
+# --description--
+
+Pass an empty callback function to the `forEach` method. For the first parameter of the callback, destructure the `author`, `image`, `url`, and `bio` items. You don't need to assign them new names – just wrap those keys in curly braces.
+
+For the second parameter, pass in `index`. This will represent the position of each author, and will be useful for pagination later.
+
+# --hints--
+
+You should destructure `author`, `image`, `url`, and `bio` as the first parameter of your callback function. Don't forget to put them inside curly braces.
+
+```js
+assert.match(code, /authors\.forEach\(\(\{author\,\s*image\,\s*url\,\s+bio\s*\}/)
+```
+
+You should destructure `index` as the second parameter of your callback function. It has to be outside of the curly braces you out the others.
+
+```js
+assert.match(code, /authors\.forEach\(\(\{.*\}\,\s*index\)/)
+```
+
+Your callback function should use an arrow syntax.
+
+```js
+assert.match(code, /authors\.forEach\(\(\{.*\}\,\s*index\)\s*=>\s*/)
+```
+
+Your callback function should be empty.
+
+```js
+assert.match(code, /authors\.forEach\(\(\{.*\}\,\s*index\)\s*=>\s*\{\n?\s*\}\);?/)
+```
```

Comment:
````suggestion
```js
assert.match(code, /authors\.forEach\(\s*\(\s*\{.*\}\s*\,\s*index\s*\)/)
```

Your callback function should use an arrow syntax.

```js
assert.match(code, /authors\.forEach\(\s*\(\s*\{.*\}\s*\,\s*index\s*\)\s*=>\s*/)
```

Your callback function should be empty.

```js
assert.match(code, /authors\.forEach\(\s*\(\s*\{.*\}\s*\,\s*index\)\s*=>\s*\{\s*\}\s*\);?/)
```
````

Allow for spacing, also `\s` matches `\n` so don't need to check that independently.